pwiz_tools/Shared/PanoramaClient/FilePicker.cs

```diff
@@ -0,0 +1,897 @@
+
+        /// <summary>
+        /// Sets a username and password and changes the 'Open' button text if a custom string is passed in
+        /// </summary>
+        private void FilePicker_Load(object sender, EventArgs e)
+        {
+            if (!string.IsNullOrEmpty(OkButtonText))
+            {
+                open.Text = OkButtonText;
+            }
+            FolderBrowser.AddFiles += AddFiles;
+            FolderBrowser.Dock = DockStyle.Fill;
+            splitContainer1.Panel1.Controls.Add(FolderBrowser);
+            FolderBrowser.NodeClick += FilePicker_MouseClick;
+        }
+
+        /// <summary>
+        /// Loads FolderBrowser and sets the state of navigation arrows
+        /// </summary>
+        public void InitializeDialog()
+        {
+            FolderBrowser = new FolderBrowser(false, ShowingSky, TreeState, Servers);
+            if (string.IsNullOrEmpty(TreeState))
+            {
+                up.Enabled = false;
+                back.Enabled = false;
+                forward.Enabled = false;
+            }
+            else
+            {
+                up.Enabled = FolderBrowser.UpEnabled();
+                back.Enabled = FolderBrowser.BackEnabled();
+                forward.Enabled = FolderBrowser.ForwardEnabled();
+            }
+            IsLoaded = true;
+        }
+
+        /// <summary>
+        /// Used for testing purposes
+        /// </summary>
+        /// <param name="serverUri"></param>
+        /// <param name="user"></param>
+        /// <param name="pass"></param>
+        /// <param name="folderJson"></param>
+        /// <param name="fileJson"></param>
+        public void InitializeTestDialog(Uri serverUri, string user, string pass, JToken folderJson, JToken fileJson)
+        {
+            _fileJson = fileJson;
+            var server = new PanoramaServer(serverUri, user, pass);
+            FolderBrowser = new FolderBrowser(server, folderJson);
+            FolderBrowser.Dock = DockStyle.Fill;
+            splitContainer1.Panel1.Controls.Add(FolderBrowser);
+            FolderBrowser.NodeClick += FilePicker_MouseClick;
+            if (string.IsNullOrEmpty(TreeState))
+            {
+                up.Enabled = false;
+                back.Enabled = false;
+                forward.Enabled = false;
+            }
+            else
+            {
+                up.Enabled = FolderBrowser.UpEnabled();
+                back.Enabled = FolderBrowser.BackEnabled();
+                forward.Enabled = FolderBrowser.ForwardEnabled();
+            }
+            IsLoaded = true;
+        }
+
+
+        /// <summary>
+        /// Builds a string that will be used as a URI to find all .sky folders
+        /// </summary>
+        /// <returns></returns>
+        private static string BuildQuery(string server, string folderPath, string queryName, string folderFilter, string[] columns, string sortParam, string equalityParam)
+        {
+            var query =
```

Comment:
I think you should use PanoramaUtil.CallNewInterface here.  Perhaps something like this? 
```
var columnsQueryParam = columns != null ? "&query.columns=" + string.Join(",", columns) : string.Empty;
var sortQueryParam = !string.IsNullOrEmpty(sortParam) ? "&query.sort={sortParam}" : string.Empty;
var allQueryParams = $"schemaName=targetedms&query.queryName={queryName}&query.containerFilterName={folderFilter}{columnsQueryParam}{sortQueryParam}";
var queryUri = PanoramaUtil.CallNewInterface(serverUri, @"query", folderPath, @"selectRows", allQueryParams);
```

I don't think `equalityParam` is used.  Can it be removed?

Reply:
I went ahead and changed the BuidQuery method based on your suggestion and removed the equalityParam that wasn't being used